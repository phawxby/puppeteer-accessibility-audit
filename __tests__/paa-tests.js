'use strict';

const auditsWithHeader = (audit, header) => audit.filter(x => x.heading === header);
let paa;

beforeEach(async () => {
  paa = require('..');
})

afterEach(async () => {
  await paa.destroy();
})

test('test that an error is thrown if not launched', async () => {
  await expect(paa.audit('__tests__/fixture.html')).rejects.toMatchSnapshot();
});

test('test that an empty URL fails', async () => {
  await paa.launch();

  await expect(paa.audit('')).rejects.toMatchSnapshot();
});

test('audit fixture.html using async', async () => {
  await paa.launch();

  let {audit, report} = await paa.audit('__tests__/fixture.html');

  expect(audit).not.toBeNull();
  expect(audit.length).not.toBeNull();
  expect(audit.length).not.toEqual(0);

  const ariaReports = auditsWithHeader(audit, 'ARIA state and property values must be valid');
  expect(ariaReports).not.toBeNull();
  expect(ariaReports.length).not.toBeNull();
  expect(ariaReports.length).toBeGreaterThan(0);
  expect(ariaReports).toMatchSnapshot();

  expect(report.length).not.toEqual(0);
});