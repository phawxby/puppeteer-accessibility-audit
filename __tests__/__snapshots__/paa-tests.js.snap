// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`audit fixture.html using async 1`] = `
Array [
  Object {
    "code": "AX_ARIA_04",
    "elements": "
DIV:nth-of-type(3) > INPUT
DIV:nth-of-type(5) > INPUT
#aria-invalid",
    "heading": "ARIA state and property values must be valid",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_04",
  },
]
`;

exports[`test that an empty URL fails 1`] = `[Error: Specify at least one URL]`;

exports[`test that an error is thrown if not launched 1`] = `[Error: You must called the \`launch\` method first to start puppeteer]`;
